### PR TITLE
Bug 1826767 - Update app to use onboarding manifest cards and enum

### PR DESCRIPTION
**Pull request by t-p-white**
- Removed JunoOnboardingPageType enum and updated uses.
- Updated app to use the card data and cards from the onboarding manifest.

Tasks:
- Removed JunoOnboardingPageType enum and updated uses to use onboarding manifest enum.
- Updated app to use the onboarding manifest card data and default cards.

### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1826767

### CODE REVIEW

File: fenix/app/src/main/java/org/mozilla/fenix/onboarding/JunoOnboardingFragment.kt [hunk 1]
```diff
@@ -111,66 +133,113 @@ class JunoOnboardingFragment : Fragment() {
...
-        buildList {
-            add(JunoOnboardingPageType.DEFAULT_BROWSER)
-            add(JunoOnboardingPageType.SYNC_SIGN_IN)
-            if (shouldShowNotificationPage(context)) {
-                add(JunoOnboardingPageType.NOTIFICATION_PERMISSION)
+    private fun pagesToDisplay(context: Context): List<OnboardingPageUiData> {
+        val allCards: Map<String, OnboardingCardData> =
+            FxNimbus.features.junoOnboarding.value().cards
+        return allCards.values
+            .filter {
```
Comment (rahulsainani) [inline fenix/app/src/main/java/org/mozilla/fenix/onboarding/JunoOnboardingFragment.kt:195]: Could remove the variable declaration and use, `FxNimbus.features.junoOnboarding.value().cards.values`, also making the function body an expression. 

I also feel we could extract this out of the fragment and add tests for this.

File: fenix/app/src/main/java/org/mozilla/fenix/onboarding/JunoOnboardingFragment.kt [hunk 2]
```diff
@@ -111,66 +133,113 @@ class JunoOnboardingFragment : Fragment() {
...
+    }
+
+    private fun OnboardingCardData.toPageUiState(): OnboardingPageUiData {
+        return when (cardType) {
+            OnboardingCardType.DEFAULT_BROWSER -> OnboardingPageUiData(
+                type = JunoOnboardingPageType.DEFAULT_BROWSER,
+                imageRes = R.drawable.ic_onboarding_welcome,
+                title = title,
+                body = body,
+                linkText = linkText!!,
```
Comment (rahulsainani) [inline fenix/app/src/main/java/org/mozilla/fenix/onboarding/JunoOnboardingFragment.kt:213]: We can remove the double bang here, both are nullable.

File: fenix/app/src/main/java/org/mozilla/fenix/onboarding/view/JunoOnboardingMapper.kt [hunk 3]
```diff
@@ -4,121 +4,64 @@
...
-        text = stringResource(id = R.string.juno_onboarding_default_browser_positive_button),
-        onClick = onPositiveButtonClick,
-    ),
-    secondaryButton = Action(
-        text = stringResource(id = R.string.juno_onboarding_default_browser_negative_button),
-        onClick = onNegativeButtonClick,
-    ),
-    onRecordImpressionEvent = {},
-)
+private fun OnboardingPageUiData.linkTextState(onUrlClick: (String) -> Unit = {}) =
```
Comment (rahulsainani) [inline fenix/app/src/main/java/org/mozilla/fenix/onboarding/view/JunoOnboardingMapper.kt:60]: Could just use this as is without extracting a function IMO

File: fenix/app/src/test/java/org/mozilla/fenix/onboarding/JunoOnboardingFragmentTest.kt [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.onboarding
+
+import org.junit.Assert.assertEquals
+import org.junit.Test
+import org.mozilla.fenix.R
+import org.mozilla.fenix.onboarding.view.OnboardingPageUiData
+
+class JunoOnboardingFragmentTest {
```
Comment (rahulsainani) [inline fenix/app/src/test/java/org/mozilla/fenix/onboarding/JunoOnboardingFragmentTest.kt:12]: I think this can be renamed to `OnboardingPageUiDataTest` as it's testing functions on List<OnboardingPageUiData> and not the fragment.

File: fenix/app/src/test/java/org/mozilla/fenix/onboarding/JunoOnboardingFragmentTest.kt [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+        val expected = "default_sync"
+        val actual = allKnownPages.take(2).telemetrySequenceId()
+
+        assertEquals(expected, actual)
+    }
+
+    @Test
+    fun `GIVEN a single known page WHEN sequenceId() called THEN should map to the correct sequence id`() {
+        val expected = "default"
+        val actual = allKnownPages.take(1).telemetrySequenceId()
```
Comment (rahulsainani) [inline fenix/app/src/test/java/org/mozilla/fenix/onboarding/JunoOnboardingFragmentTest.kt:49]: Would recommend
```
val actual = listOf(default).telemetrySequenceId()
```
so there is no extra logic in the code path of the test. So we purely test the prod code. 

Same for the test above.

File: fenix/app/src/main/java/org/mozilla/fenix/onboarding/view/OnboardingPageState.kt [hunk 6]
```diff
@@ -24,7 +24,7 @@ data class OnboardingPageState(
     val linkTextState: LinkTextState? = null,
     val primaryButton: Action,
     val secondaryButton: Action? = null,
-    val onRecordImpressionEvent: () -> Unit,
+    val onRecordImpressionEvent: (() -> Unit)? = null,
```
Comment (rahulsainani) [inline fenix/app/src/main/java/org/mozilla/fenix/onboarding/view/OnboardingPageState.kt:27]: Let's not change the component for this PR

File: fenix/app/src/main/java/org/mozilla/fenix/onboarding/view/OnboardingPage.kt [hunk 7]
```diff
@@ -148,7 +148,7 @@ fun OnboardingPage(
             }
 
             LaunchedEffect(pageState) {
-                pageState.onRecordImpressionEvent()
+                pageState.onRecordImpressionEvent
```
Comment (rahulsainani) [inline fenix/app/src/main/java/org/mozilla/fenix/onboarding/view/OnboardingPage.kt:151]: This will cause a bug for the features which use this component to call this lambda. Let's revert this change.

File: fenix/app/src/androidTest/java/org/mozilla/fenix/onboarding/view/JunoOnboardingMapperTest.kt [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+    body = StringHolder(null, "notification body"),
+    primaryButtonLabel = StringHolder(null, "notification primary button text"),
+    secondaryButtonLabel = StringHolder(null, "notification secondary button text"),
+    ordering = 30,
+)
+
+private val allKnownCardData = listOf(
+    defaultBrowserCardData,
+    syncCardData,
+    notificationCardData,
```
Comment (rahulsainani) [inline fenix/app/src/androidTest/java/org/mozilla/fenix/onboarding/view/JunoOnboardingMapperTest.kt:86]: 💡 More of an idea really, Should we update the list order here so the sortedBy { ordering } is tested as well?

**Review by rahulsainani — APPROVED** [review]
Looking good! Thanks for updating the PR and for all the discussions around refactoring and testing 🚀

File: fenix/app/src/androidTest/java/org/mozilla/fenix/onboarding/view/JunoOnboardingMapperTest.kt (hunk 8 above)
Comment (t-p-white) [reply fenix/app/src/androidTest/java/org/mozilla/fenix/onboarding/view/JunoOnboardingMapperTest.kt:86]: Good suggestion, changed ordering 👍
